lib/dash/mpd_parser.js

```diff
@@ -50,8 +51,12 @@ shaka.dash.mpd.parseMpd = function(source, url) {
     return null;
   }
 
+  if (!baseUrl) {
```

Comment:
You should drop this condition, since it should not be possible after successful compilation.  Otherwise, the PR looks good to me.